app/assets/v2/js/grants/new_match.js

```diff
@@ -0,0 +1,188 @@
+Vue.component('v-select', VueSelect.VueSelect);
+
+
+Vue.mixin({
+  methods: {
+    transfer_web3: function(params) {
+      let vm = this;
+
+      const amount = params.amount;
+      const decimals = 18;
+      const to = '0xde21F729137C5Af1b01d73aF1dC21eFfa2B8a0d6';
+      // const token_addr = "0x6B175474E89094C44Da98b954EedeAC495271d0F";
+      const token_addr = (document.web3network == 'mainnet') ?
+        '0x6B175474E89094C44Da98b954EedeAC495271d0F' :
+        '0x5592EC0cfb4dbc12D3aB100b257153436a1f0FEa'
+
+      ;
+      const amountInWei = amount * 1.0 * Math.pow(10, decimals);
+      const amountAsString = new web3.utils.BN(BigInt(amountInWei)).toString();
+
+      const token_contract = new web3.eth.Contract(token_abi, token_addr);
+
+      return token_contract.methods.transfer(to, web3.utils.toHex(amountAsString))
+        .send({from: selectedAccount},
+          (error, tx_id) => {
+            if (error) {
+              _alert({ message: gettext('Unable to pay the match pledge amount. Please try again.') }, 'error');
+              console.error(`error: unable to pay pledge due to : ${error}`);
+              return;
+            }
+            params['tx_id'] = tx_id;
+            console.log(params);
+            vm.submitted = true;
+            vm.createMatchingPledge(params);
+
+          }
+        );
+    },
+    checkForm: function(e) {
+      let vm = this;
+
+      vm.submitted = true;
+      vm.errors = {};
+      if (!vm.form.why.length) {
+        vm.$set(vm.errors, 'why', 'Please select why you want to create a match pledge');
+      }
+      if (!vm.form.amount) {
+        vm.$set(vm.errors, 'amount', 'Please enter amount you would want to pledge');
+      }
+      if (!vm.form.stage) {
+        vm.$set(vm.errors, 'stage', 'Please select the stage of funding');
+      }
+
+      if (!vm.grants_to_fund) {
+        vm.$set(vm.errors, 'grants_to_fund', 'Please select one of the options');
+      } else if (vm.grants_to_fund == 'types' && !vm.form.grant_types.length > 0) {
+        vm.$set(vm.errors, 'grant_types', 'Please select the grant types');
+      } else if (vm.grants_to_fund == 'collections' && !vm.form.grant_collections.length > 0) {
+        vm.$set(vm.errors, 'grant_collections', 'Please select the collections');
+      }
+
+
+      if (Object.keys(vm.errors).length) {
+        return false; // there are errors the user must correct
+      }
+      vm.submitted = false;
+      return true; // no errors, continue to create match pledge
+    },
+    submitForm: async function(event) {
```

Comment:
I think you need to trigger the wallet if the user is not connected yet

Reply:
you mean this piece of code ?
```
 if (!provider) {
        onConnect();
        return false;
      }
```